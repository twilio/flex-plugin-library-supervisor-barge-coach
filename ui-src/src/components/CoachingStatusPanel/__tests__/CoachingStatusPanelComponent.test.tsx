import React from 'react';
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event';
import BargeCoachService from '../../../utils/serverless/BargeCoachService';
import { CoachingStatusPanel } from '../CoachingStatusPanelComponent';

jest.mock('../../../utils/serverless/BargeCoachService');
jest.mock('react-redux', () => ({
  useSelector: () => ({
    supervisorArray:[
        {
            status:'is Monitoring'
        }
    ],
    syncSubscribed:false
  }),
  useDispatch: () => jest.fn()
}));
jest.mock('../../../utils/sync/Sync', () => ({
    SyncDoc:{
        getSyncDoc:()=>Promise.resolve({
            data:{
                supervisors:[
                    {
                        status:'is Monitoring'
                    }
                ]
            },
            on:jest.fn()
        })
    }
}))

describe('Coaching status panel',()=>{
    it('renders correct snapshot',()=>{
        const wrapper = render(
            <CoachingStatusPanel {...{}} />
          )
          expect(wrapper).toMatchSnapshot();
    })
})