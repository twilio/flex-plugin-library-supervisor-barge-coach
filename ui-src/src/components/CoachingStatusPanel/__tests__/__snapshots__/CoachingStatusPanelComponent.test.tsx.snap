// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coaching status panel renders correct snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <style>
      
    table.disable-selection {
      -webkit-touch-callout: none;
      -webkit-user-select: none; 
      -khtml-user-select: none; 
      -moz-user-select: none; 
      -ms-user-select: none; 
      user-select: none;
    }
  
    .disable-selection span::selection{
      background-color: transparent;
    }
    .disable-selection br::selection{
      background-color: transparent;
    }
  
    </style>
    <div>
      <div
        class="css-x2lkna"
        data-paste-element="FLEX"
      >
        <div
          class="css-tfro19"
          data-paste-element="COACH_STATUS_PANEL_BOX"
        >
          <div
            class="css-roynbj"
            data-paste-element="COACH_STATUS_PANEL_BOX_CHILD"
          >
            <div
              class="css-176snxq"
              data-paste-element="BOX"
            >
              You are being Coached by:
              <div
                class="css-roynbj"
                data-paste-element="BOX"
              >
                <ol>
                  <p
                    class="css-17d3dsd"
                    data-paste-element="TEXT"
                  >
                    <li />
                  </p>
                </ol>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-x2lkna"
      data-paste-element="FLEX"
    >
      <div
        class="css-tfro19"
        data-paste-element="COACH_STATUS_PANEL_BOX"
      >
        <div
          class="css-roynbj"
          data-paste-element="COACH_STATUS_PANEL_BOX_CHILD"
        >
          <div
            class="css-176snxq"
            data-paste-element="BOX"
          >
            You are being Coached by:
            <div
              class="css-roynbj"
              data-paste-element="BOX"
            >
              <ol>
                <p
                  class="css-17d3dsd"
                  data-paste-element="TEXT"
                >
                  <li />
                </p>
              </ol>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
